### PR TITLE
PostgresApp and Homebrew Postgres on MacOS

### PR DESCRIPTION
**Pull request by noelmcloughlin**
Support `PostgresApp` and `Homebrew Postgres` on MacOS.  

*** POSTGRESAPP ***
```
local:
----------
          ID: pg-remove-prev-archive
    Function: file.absent
        Name: /tmp/postgrestmp/postgres.dmg
      Result: True
     Comment: Removed file /tmp/postgrestmp/postgres.dmg
     Started: 10:26:44.473380
    Duration: 11.411 ms
     Changes:   
              ----------
              removed:
                  /tmp/postgrestmp/postgres.dmg
----------
          ID: pg-extract-dirs
    Function: file.directory
        Name: /tmp/postgrestmp
      Result: True
     Comment: Directory /tmp/postgrestmp is in the correct state
              Directory /tmp/postgrestmp updated
     Started: 10:26:44.484917
    Duration: 0.752 ms
     Changes:   
----------
          ID: pg-download-archive
    Function: pkg.installed
        Name: curl
      Result: True
     Comment: Package curl is already installed
     Started: 10:26:46.269667
    Duration: 733.215 ms
     Changes:   
----------
          ID: pg-download-archive
    Function: cmd.run
        Name: curl -s -L -o '/tmp/postgrestmp/postgres.dmg' https://github.com/PostgresApp/PostgresApp/releases/download/v2.1.1/Postgres-2.1.1.dmg
      Result: True
     Comment: Command "curl -s -L -o '/tmp/postgrestmp/postgres.dmg' https://github.com/PostgresApp/PostgresApp/releases/download/v2.1.1/Postgres-2.1.1.dmg" run
     Started: 10:26:47.005116
    Duration: 229067.193 ms
     Changes:   
              ----------
              pid:
                  5116
              retcode:
                  0
              stderr:
              stdout:
----------
          ID: pg-package-install
    Function: macpackage.installed
        Name: /tmp/postgrestmp/postgres.dmg
      Result: True
     Comment: Postgres.app installed
     Started: 10:30:36.295101
    Duration: 18240.587 ms
     Changes:   
              ----------
              installed:
                  - Postgres.app
----------
          ID: pg-package-install
    Function: file.append
        Name: /Users/messi/.bash_profile
      Result: True
     Comment: Appended 1 lines
     Started: 10:30:54.536223
    Duration: 20.205 ms
     Changes:   
              ----------
              diff:
                  --- 
                  
                  +++ 
                  
                  @@ -26,3 +26,4 @@
                  
                   #### HOSTS #####
                   alias tm='cd && ssh tm'
                   export PATH="/usr/local/sbin:$PATH"
                  +export PATH=$PATH:/Applications/Postgres.app/Contents/Versions/latest/bin
----------
          ID: pg-remove-archive
    Function: file.absent
        Name: /tmp/postgrestmp
      Result: True
     Comment: Removed directory /tmp/postgrestmp
     Started: 10:30:54.556690
    Duration: 2.163 ms
     Changes:   
              ----------
              removed:
                  /tmp/postgrestmp
----------
          ID: postgres_maxfiles_limits_conf
    Function: file.managed
        Name: /Library/LaunchDaemons/limit.maxfiles.plist
      Result: True
     Comment: File /Library/LaunchDaemons/limit.maxfiles.plist is in the correct state
     Started: 10:30:54.558930
    Duration: 13.938 ms
     Changes:   
----------
          ID: postgres-desktop-shortcut-clean
    Function: file.absent
        Name: /Users/messi/Desktop/postgres
      Result: True
     Comment: File /Users/messi/Desktop/postgres is not present
     Started: 10:30:54.572969
    Duration: 0.427 ms
     Changes:   
----------
          ID: postgres-desktop-shortcut-add
    Function: file.managed
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: File /tmp/mac_shortcut.sh updated
     Started: 10:30:54.573711
    Duration: 7.17 ms
     Changes:   
              ----------
              diff:
                  New file
              mode:
                  0755
              user:
                  messi
----------
          ID: postgres-desktop-shortcut-add
    Function: cmd.run
        Name: /tmp/mac_shortcut.sh postgresapp
      Result: True
     Comment: Command "/tmp/mac_shortcut.sh postgresapp" run
     Started: 10:30:54.581287
    Duration: 440.299 ms
     Changes:   
              ----------
              pid:
                  5168
              retcode:
                  0
              stderr:
              stdout:
                  alias file Postgres of folder Desktop of folder messi of folder Users of startup disk

Summary for local
-------------
Succeeded: 11 (changed=7)
Failed:     0
-------------
Total states run:     11
Total run time:  248.537 s

```

** HOMEBREW **
```
[ERROR   ] Service homebrew.mxcl.postgresql failed to start
local:
----------
          ID: postgresql-server
    Function: pkg.installed
      Result: True
     Comment: The following packages were installed/updated: postgresql
     Started: 13:13:12.225375
    Duration: 6148.853 ms
     Changes:   
              ----------
              postgresql:
                  ----------
                  new:
                      10.1
                  old:
----------
          ID: postgresql-server
    Function: file.managed
        Name: /Library/LaunchAgents/homebrew.mxcl.postgresql.plist
      Result: True
     Comment: File /Library/LaunchAgents/homebrew.mxcl.postgresql.plist updated
     Started: 13:13:13.378074
    Duration: 13.146 ms
     Changes:   
              ----------
              diff:
                  New file
              mode:
                  0644
----------
          ID: postgresql-cluster-prepared
    Function: cmd.run
        Name: initdb -D /Users/messi/Library/AppSupport/postgres_homebrew
      Result: True
     Comment: Command "initdb -D /Users/messi/Library/AppSupport/postgres_homebrew" run
     Started: 13:13:13.392806
    Duration: 801.939 ms
     Changes:   
              ----------
              pid:
                  30633
              retcode:
                  0
              stderr:
                  
                  WARNING: enabling "trust" authentication for local connections
                  You can change this by editing pg_hba.conf or using the option -A, or
                  --auth-local and --auth-host, the next time you run initdb.
              stdout:
                  The files belonging to this database system will be owned by user "messi".
                  This user must also own the server process.
                  
                  The database cluster will be initialized with locale "C".
                  The default database encoding has accordingly been set to "SQL_ASCII".
                  The default text search configuration will be set to "english".
                  
                  Data page checksums are disabled.
                  
                  creating directory /Users/messi/Library/AppSupport/postgres_homebrew ... ok
                  creating subdirectories ... ok
                  selecting default max_connections ... 100
                  selecting default shared_buffers ... 128MB
                  selecting dynamic shared memory implementation ... posix
                  creating configuration files ... ok
                  running bootstrap script ... ok
                  performing post-bootstrap initialization ... ok
                  syncing data to disk ... ok
                  
                  Success. You can now start the database server using:
                  
                      pg_ctl -D /Users/messi/Library/AppSupport/postgres_homebrew -l logfile start
----------
          ID: postgresql-config-dir
    Function: file.directory
        Name: /Users/messi/Library/AppSupport/postgres_homebrew
      Result: True
     Comment: Directory /Users/messi/Library/AppSupport/postgres_homebrew updated
     Started: 13:13:19.195268
    Duration: 219.31 ms
     Changes:   
              ----------
              mode:
                  0775
----------
          ID: postgresql-conf
    Function: file.blockreplace
        Name: /Users/messi/Library/AppSupport/postgres_homebrew/postgresql.conf
      Result: True
     Comment: Changes were made
     Started: 13:13:19.414895
    Duration: 4.529 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -656,3 +656,6 @@
                   #------------------------------------------------------------------------------
                   
                   # Add settings for extensions here
                  +# Managed by SaltStack: listen_addresses: please do not edit
                  +listen_addresses = '*'  # listen on all interfaces
                  +# Managed by SaltStack: end of salt managed zone --
----------
          ID: postgresql-pg_hba
    Function: file.managed
        Name: /Users/messi/Library/AppSupport/postgres_homebrew/pg_hba.conf
      Result: True
     Comment: File /Users/messi/Library/AppSupport/postgres_homebrew/pg_hba.conf updated
     Started: 13:13:19.419582
    Duration: 3792.969 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1,93 +1,23 @@
                  +######################################################################
                  +# ATTENTION! Managed by SaltStack.                                   #
                  +# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN! #
                  +######################################################################
                  +#
                   # PostgreSQL Client Authentication Configuration File
                   # ===================================================
                   #
                   # Refer to the "Client Authentication" section in the PostgreSQL
                  -# documentation for a complete description of this file.  A short
                  -# synopsis follows.
                  -#
                  -# This file controls: which hosts are allowed to connect, how clients
                  -# are authenticated, which PostgreSQL user names they can use, which
                  -# databases they can access.  Records take one of these forms:
                  -#
                  -# local      DATABASE  USER  METHOD  [OPTIONS]
                  -# host       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -# hostssl    DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -# hostnossl  DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -#
                  -# (The uppercase items must be replaced by actual values.)
                  -#
                  -# The first field is the connection type: "local" is a Unix-domain
                  -# socket, "host" is either a plain or SSL-encrypted TCP/IP socket,
                  -# "hostssl" is an SSL-encrypted TCP/IP socket, and "hostnossl" is a
                  -# plain TCP/IP socket.
                  -#
                  -# DATABASE can be "all", "sameuser", "samerole", "replication", a
                  -# database name, or a comma-separated list thereof. The "all"
                  -# keyword does not match "replication". Access to replication
                  -# must be enabled in a separate record (see example below).
                  -#
                  -# USER can be "all", a user name, a group name prefixed with "+", or a
                  -# comma-separated list thereof.  In both the DATABASE and USER fields
                  -# you can also write a file name prefixed with "@" to include names
                  -# from a separate file.
                  -#
                  -# ADDRESS specifies the set of hosts the record matches.  It can be a
                  -# host name, or it is made up of an IP address and a CIDR mask that is
                  -# an integer (between 0 and 32 (IPv4) or 128 (IPv6) inclusive) that
                  -# specifies the number of significant bits in the mask.  A host name
                  -# that starts with a dot (.) matches a suffix of the actual host name.
                  -# Alternatively, you can write an IP address and netmask in separate
                  -# columns to specify the set of hosts.  Instead of a CIDR-address, you
                  -# can write "samehost" to match any of the server's own IP addresses,
                  -# or "samenet" to match any address in any subnet that the server is
                  -# directly connected to.
                  -#
                  -# METHOD can be "trust", "reject", "md5", "password", "scram-sha-256",
                  -# "gss", "sspi", "ident", "peer", "pam", "ldap", "radius" or "cert".
                  -# Note that "password" sends passwords in clear text; "md5" or
                  -# "scram-sha-256" are preferred since they send encrypted passwords.
                  -#
                  -# OPTIONS are a set of options for the authentication in the format
                  -# NAME=VALUE.  The available options depend on the different
                  -# authentication methods -- refer to the "Client Authentication"
                  -# section in the documentation for a list of which options are
                  -# available for which authentication methods.
                  -#
                  -# Database and user names containing spaces, commas, quotes and other
                  -# special characters must be quoted.  Quoting one of the keywords
                  -# "all", "sameuser", "samerole" or "replication" makes the name lose
                  -# its special character, and just match a database or username with
                  -# that name.
                  -#
                  -# This file is read on server startup and when the server receives a
                  -# SIGHUP signal.  If you edit the file on a running system, you have to
                  -# SIGHUP the server for the changes to take effect, run "pg_ctl reload",
                  -# or execute "SELECT pg_reload_conf()".
                  -#
                  -# Put your actual configuration here
                  -# ----------------------------------
                  -#
                  -# If you want to allow non-local connections, you need to add more
                  -# "host" records.  In that case you will also need to make PostgreSQL
                  -# listen on a non-local interface via the listen_addresses
                  -# configuration parameter, or via the -i or -h command line switches.
                  +# documentation for a complete description of this file.
                   
                  -# CAUTION: Configuring the system for local "trust" authentication
                  -# allows any local user to connect as any PostgreSQL user, including
                  -# the database superuser.  If you do not trust all your local users,
                  -# use another authentication method.
                  +# DO NOT DISABLE!
                  +# If you change this first entry you will need to make sure that the
                  +# database superuser can access the database using some other method.
                  +# Noninteractive access to all databases is required during automatic
                  +# maintenance (custom daily cronjobs, replication, and similar tasks).
                   
                  +# Database administrative login by Unix domain socket
                  +local   all             postgres                                peer
                   
                   # TYPE  DATABASE        USER            ADDRESS                 METHOD
                  -
                  -# "local" is for Unix domain socket connections only
                  -local   all             all                                     trust
                  -# IPv4 local connections:
                  -host    all             all             127.0.0.1/32            trust
                  -# IPv6 local connections:
                  -host    all             all             ::1/128                 trust
                  -# Allow replication connections from localhost, by a user with the
                  -# replication privilege.
                  -local   replication     all                                     trust
                  -host    replication     all             127.0.0.1/32            trust
                  -host    replication     all             ::1/128                 trust
                  +local   db1             localUser                               md5
                  +host    db2             remoteUser      192.168.33.0/24         md5
----------
          ID: postgresql-tablespace-dir-my_space
    Function: file.directory
        Name: /srv/my_tablespace
      Result: True
     Comment: Directory /srv/my_tablespace is in the correct state
              Directory /srv/my_tablespace updated
     Started: 13:13:23.212795
    Duration: 47.959 ms
     Changes:   
----------
          ID: postgresql-running
    Function: service.running
        Name: homebrew.mxcl.postgresql
      Result: True
     Comment: Service started
     Started: 13:13:23.296112
    Duration: 17.441 ms
     Changes:   
              ----------
              homebrew.mxcl.postgresql:
                  True
----------
          ID: postgresql-client-libs
    Function: pkg.installed
      Result: True
     Comment: No packages to install provided
     Started: 13:13:23.313812
    Duration: 0.447 ms
     Changes:   
----------
          ID: postgres_maxfiles_limits_conf
    Function: file.managed
        Name: /Library/LaunchDaemons/limit.maxfiles.plist
      Result: True
     Comment: File /Library/LaunchDaemons/limit.maxfiles.plist is in the correct state
     Started: 13:13:23.314350
    Duration: 2.801 ms
     Changes:   

Summary for local
-------------
Succeeded: 10 (changed=7)
Failed:     0
-------------
Total states run:     10
Total run time:   11.049 s
```

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
**UBUNTU** No regression.

```
         ID: postgresql-server
    Function: pkg.installed
      Result: True
     Comment: 2 targeted packages were installed/updated.
     Started: 13:35:30.079385
    Duration: 14040.802 ms
     Changes:   
              ----------
              postgresql-9.5:
                  ----------
                  new:
                      9.5.12-0ubuntu0.16.04
                  old:
              postgresql-client:
                  ----------
                  new:
                      1
                  old:
              postgresql-client-9.5:
                  ----------
                  new:
                      9.5.12-0ubuntu0.16.04
                  old:
              postgresql-client-common:
                  ----------
                  new:
                      173ubuntu0.1
                  old:
              postgresql-common:
                  ----------
                  new:
                      173ubuntu0.1
                  old:
              postgresql-contrib:
                  ----------
                  new:
                      9.5+173ubuntu0.1
                  old:
              postgresql-contrib-9.5:
                  ----------
                  new:
                      9.5.12-0ubuntu0.16.04
                  old:
----------
          ID: postgresql-cluster-prepared
    Function: file.directory
        Name: /var/lib/postgresql/9.5/main
      Result: True
     Comment: onlyif execution failed
     Started: 13:35:44.128900
    Duration: 1646.611 ms
     Changes:   
----------
          ID: postgresql-cluster-prepared
    Function: cmd.run
        Name: pg_createcluster 9.5 main -d  /var/lib/postgresql/9.5/main
      Result: True
     Comment: Command "pg_createcluster 9.5 main -d  /var/lib/postgresql/9.5/main" run
     Started: 13:35:45.780753
    Duration: 3808.583 ms
     Changes:   
              ----------
              pid:
                  2045
              retcode:
                  0
              stderr:
              stdout:
                  Creating new cluster 9.5/main ...
                    config /etc/postgresql/9.5/main
                    data   /var/lib/postgresql/9.5/main
                    locale C
                    socket /var/run/postgresql
                    port   5432
----------
          ID: postgresql-config-dir
    Function: file.directory
        Name: /var/lib/postgresql/9.5/main
      Result: True
     Comment: Directory /var/lib/postgresql/9.5/main is in the correct state
              Directory /var/lib/postgresql/9.5/main updated
     Started: 13:35:49.591459
    Duration: 623.862 ms
     Changes:   
----------
          ID: postgresql-config-dir
    Function: file.directory
        Name: /etc/postgresql/9.5/main
      Result: True
     Comment: Directory /etc/postgresql/9.5/main updated
     Started: 13:35:50.216524
    Duration: 5.047 ms
     Changes:   
              ----------
              mode:
                  0700
----------
          ID: postgresql-conf
    Function: file.blockreplace
        Name: /etc/postgresql/9.5/main/postgresql.conf
      Result: True
     Comment: Changes were made
     Started: 13:35:50.223050
    Duration: 7.272 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -623,3 +623,6 @@
                   #------------------------------------------------------------------------------
                   
                   # Add settings for extensions here
                  +# Managed by SaltStack: listen_addresses: please do not edit
                  +listen_addresses = '*'  # listen on all interfaces
                  +# Managed by SaltStack: end of salt managed zone --
----------
          ID: postgresql-pg_hba
    Function: file.managed
        Name: /etc/postgresql/9.5/main/pg_hba.conf
      Result: True
     Comment: File /etc/postgresql/9.5/main/pg_hba.conf updated
     Started: 13:35:50.231545
    Duration: 42316.68 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1,99 +1,23 @@
                  +######################################################################
                  +# ATTENTION! Managed by SaltStack.                                   #
                  +# DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN! #
                  +######################################################################
                  +#
                   # PostgreSQL Client Authentication Configuration File
                   # ===================================================
                   #
                   # Refer to the "Client Authentication" section in the PostgreSQL
                  -# documentation for a complete description of this file.  A short
                  -# synopsis follows.
                  -#
                  -# This file controls: which hosts are allowed to connect, how clients
                  -# are authenticated, which PostgreSQL user names they can use, which
                  -# databases they can access.  Records take one of these forms:
                  -#
                  -# local      DATABASE  USER  METHOD  [OPTIONS]
                  -# host       DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -# hostssl    DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -# hostnossl  DATABASE  USER  ADDRESS  METHOD  [OPTIONS]
                  -#
                  -# (The uppercase items must be replaced by actual values.)
                  -#
                  -# The first field is the connection type: "local" is a Unix-domain
                  -# socket, "host" is either a plain or SSL-encrypted TCP/IP socket,
                  -# "hostssl" is an SSL-encrypted TCP/IP socket, and "hostnossl" is a
                  -# plain TCP/IP socket.
                  -#
                  -# DATABASE can be "all", "sameuser", "samerole", "replication", a
                  -# database name, or a comma-separated list thereof. The "all"
                  -# keyword does not match "replication". Access to replication
                  -# must be enabled in a separate record (see example below).
                  -#
                  -# USER can be "all", a user name, a group name prefixed with "+", or a
                  -# comma-separated list thereof.  In both the DATABASE and USER fields
                  -# you can also write a file name prefixed with "@" to include names
                  -# from a separate file.
                  -#
                  -# ADDRESS specifies the set of hosts the record matches.  It can be a
                  -# host name, or it is made up of an IP address and a CIDR mask that is
                  -# an integer (between 0 and 32 (IPv4) or 128 (IPv6) inclusive) that
                  -# specifies the number of significant bits in the mask.  A host name
                  -# that starts with a dot (.) matches a suffix of the actual host name.
                  -# Alternatively, you can write an IP address and netmask in separate
                  -# columns to specify the set of hosts.  Instead of a CIDR-address, you
                  -# can write "samehost" to match any of the server's own IP addresses,
                  -# or "samenet" to match any address in any subnet that the server is
                  -# directly connected to.
                  -#
                  -# METHOD can be "trust", "reject", "md5", "password", "gss", "sspi",
                  -# "ident", "peer", "pam", "ldap", "radius" or "cert".  Note that
                  -# "password" sends passwords in clear text; "md5" is preferred since
                  -# it sends encrypted passwords.
                  -#
                  -# OPTIONS are a set of options for the authentication in the format
                  -# NAME=VALUE.  The available options depend on the different
                  -# authentication methods -- refer to the "Client Authentication"
                  -# section in the documentation for a list of which options are
                  -# available for which authentication methods.
                  -#
                  -# Database and user names containing spaces, commas, quotes and other
                  -# special characters must be quoted.  Quoting one of the keywords
                  -# "all", "sameuser", "samerole" or "replication" makes the name lose
                  -# its special character, and just match a database or username with
                  -# that name.
                  -#
                  -# This file is read on server startup and when the postmaster receives
                  -# a SIGHUP signal.  If you edit the file on a running system, you have
                  -# to SIGHUP the postmaster for the changes to take effect.  You can
                  -# use "pg_ctl reload" to do that.
                  -
                  -# Put your actual configuration here
                  -# ----------------------------------
                  -#
                  -# If you want to allow non-local connections, you need to add more
                  -# "host" records.  In that case you will also need to make PostgreSQL
                  -# listen on a non-local interface via the listen_addresses
                  -# configuration parameter, or via the -i or -h command line switches.
                  -
                  -
                  -
                  +# documentation for a complete description of this file.
                   
                   # DO NOT DISABLE!
                   # If you change this first entry you will need to make sure that the
                   # database superuser can access the database using some other method.
                   # Noninteractive access to all databases is required during automatic
                   # maintenance (custom daily cronjobs, replication, and similar tasks).
                  -#
                  +
                   # Database administrative login by Unix domain socket
                   local   all             postgres                                peer
                   
                   # TYPE  DATABASE        USER            ADDRESS                 METHOD
                  -
                  -# "local" is for Unix domain socket connections only
                  -local   all             all                                     peer
                  -# IPv4 local connections:
                  -host    all             all             127.0.0.1/32            md5
                  -# IPv6 local connections:
                  -host    all             all             ::1/128                 md5
                  -# Allow replication connections from localhost, by a user with the
                  -# replication privilege.
                  -#local   replication     postgres                                peer
                  -#host    replication     postgres        127.0.0.1/32            md5
                  -#host    replication     postgres        ::1/128                 md5
                  +local   db1             localUser                               md5
                  +host    db2             remoteUser      192.168.33.0/24         md5
              mode:
                  0600
----------
          ID: postgresql-tablespace-dir-my_space
    Function: file.directory
        Name: /srv/my_tablespace
      Result: True
     Comment: Directory /srv/my_tablespace updated
     Started: 13:36:32.549754
    Duration: 2.245 ms
     Changes:   
              ----------
              /srv/my_tablespace:
                  New Dir
----------
          ID: postgresql-running
    Function: service.running
        Name: postgresql
      Result: True
     Comment: Service postgresql has been enabled, and is running
     Started: 13:36:32.554226
    Duration: 3258.194 ms
     Changes:   
              ----------
              postgresql:
                  True
----------
          ID: postgresql-client-libs
    Function: pkg.installed
      Result: True
     Comment: The following packages were installed/updated: libpq-dev
              The following packages were already installed: postgresql-client-9.5
     Started: 13:36:35.812941
    Duration: 7050.202 ms
     Changes:   
              ----------
              libpq-dev:
                  ----------
                  new:
                      9.5.12-0ubuntu0.16.04
                  old:
----------
          ID: postgres-reload-modules
    Function: test.succeed_with_changes
      Result: True
     Comment: Success!
     Started: 13:36:42.866458
    Duration: 53.651 ms
     Changes:   
              ----------
              testing:
                  ----------
                  new:
                      Something pretended to change
                  old:
                      Unchanged
----------
          ID: postgres_user-absentUser
    Function: postgres_user.absent
        Name: absentUser
      Result: True
     Comment: User absentUser is not present, so it cannot be removed
     Started: 13:36:42.930660
    Duration: 566.452 ms
     Changes:   
----------
          ID: postgres_user-localUser
    Function: postgres_user.present
        Name: localUser
      Result: True
     Comment: The user localUser has been created
     Started: 13:36:43.499792
    Duration: 1438.53 ms
     Changes:   
              ----------
              localUser:
                  Present
----------
          ID: postgres_user-remoteUser
    Function: postgres_user.present
        Name: remoteUser
      Result: True
     Comment: The user remoteUser has been created
     Started: 13:36:44.940962
    Duration: 1864.598 ms
     Changes:   
              ----------
              remoteUser:
                  Present
----------
          ID: postgres_tablespace-my_space
    Function: postgres_tablespace.present
        Name: my_space
      Result: True
     Comment: The tablespace my_space has been created
     Started: 13:36:46.809347
    Duration: 309.056 ms
     Changes:   
              ----------
              my_space:
                  Present
----------
          ID: postgres_database-db1
    Function: postgres_database.present
        Name: db1
      Result: True
     Comment: The database db1 has been created
     Started: 13:36:47.128311
    Duration: 1044.444 ms
     Changes:   
              ----------
              db1:
                  Present
----------
          ID: postgres_database-db2
    Function: postgres_database.present
        Name: db2
      Result: True
     Comment: The database db2 has been created
     Started: 13:36:48.179775
    Duration: 1082.117 ms
     Changes:   
              ----------
              db2:
                  Present
    Warnings: 'extensions' and 'schemas' are invalid keyword arguments for
              'postgres_database.present'. If you were trying to pass additional
              data to be used in a template context, please populate 'context'
              with 'key: value' pairs. Your approach will work until Salt
              Fluorine is out. Please update your state files.
----------
          ID: postgres_schema-uuid_ossp
    Function: postgres_schema.present
        Name: uuid_ossp
      Result: True
     Comment: Schema uuid_ossp has been created in database db1
     Started: 13:36:49.267540
    Duration: 618.166 ms
     Changes:   
              ----------
              uuid_ossp:
                  Present
----------
          ID: postgres_extension-uuid-ossp
    Function: postgres_extension.present
        Name: uuid-ossp
      Result: True
     Comment: The extension uuid-ossp has been installed
     Started: 13:36:49.893800
    Duration: 1058.071 ms
     Changes:   
              ----------
              uuid-ossp:
                  Installed
```

File: postgres/templates/limit.maxfiles.plist [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>  
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"  
+        "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">  
+  <dict>
+    <key>Label</key>
+    <string>limit.maxfiles</string>
+    <key>ProgramArguments</key>
+    <array>
+      <string>launchctl</string>
```
Comment (weswhet) [inline postgres/templates/limit.maxfiles.plist]: Probably best to do the full path here. `/bin/launchctl`.

**Comment by weswhet** [comment]
Other than the one comment I made this, LGTM.

File: postgres/templates/limit.maxfiles.plist (hunk 1 above)
Comment (noelmcloughlin) [reply postgres/templates/limit.maxfiles.plist]: Thank you for reviewing @weswhet,  Good spot, I fixed that issue!!

**Pull request merged**